File: DittoDatto-old/services/mercury-engine/src/mercury_core/models/category.py
"""Category model for platform-wide service taxonomy.

Stored in titan/discovery.category.
Admin-managed. Used for discovery filtering and DittoBar search.

Schema source: schemas/discovery.surql §1 (category table)
"""

from __future__ import annotations

from mercury_core.models.common import MercuryModel, TimestampMixin


class Category(MercuryModel, TimestampMixin):
    """Platform-wide service category (taxonomy).

    Categories classify establishments for discovery.
    Managed by platform admins, not company operators.

    Attributes:
        name: Display name (e.g., "Hair Salon", "Thai Massage").
        slug: URL-safe identifier (unique index in SurrealDB).
        description: Optional long description.
        icon: Optional icon identifier (e.g., Material icon name).
        count: Number of establishments in this category (synced).
    """

    id: str | None = None
    name: str
    slug: str
    description: str | None = None
    icon: str | None = None
    count: int = 0

File: DittoDatto-old/packages/mercury_client/lib/src/models/category.dart
import 'package:json_annotation/json_annotation.dart';

part 'category.g.dart';

/// Platform-wide service category (taxonomy).
///
/// Mirrors: mercury_core/models/category.py
/// Stored in: titan/discovery.category
@JsonSerializable(fieldRename: FieldRename.snake)
class Category {
  const Category({
    this.id,
    required this.name,
    required this.slug,
    this.description,
    this.icon,
    this.count = 0,
    this.createdAt,
    this.updatedAt,
  });

  final String? id;
  final String name;
  final String slug;
  final String? description;
  final String? icon;
  final int count;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  factory Category.fromJson(Map<String, dynamic> json) =>
      _$CategoryFromJson(json);
  Map<String, dynamic> toJson() => _$CategoryToJson(this);
}
